includes/checkout/template.php

```diff
@@ -1099,5 +1099,30 @@ function edd_receipt_show_download_files( $item_id, $receipt_args, $item = array
 		}
 	}
 
-	return apply_filters( 'edd_receipt_show_download_files', $ret, $item_id, $receipt_args, $item );
+	if ( has_filter( 'edd_receipt_show_download_files' ) ) {
+		$cart = array();
+		$item = $order_item;
+		if ( ! empty( $order_item->order_id ) ) {
+			$order = edd_get_order_by( 'id', $order_item->order_id );
+			$cart  = edd_get_payment_meta_cart_details( $order->id, true );
+			$item  = $cart[ $item->cart_index ];
+		}
+		$ret = apply_filters( 'edd_receipt_show_download_files', $ret, $item_id, $receipt_args, $item );
+	}
+
+	// If the $order_item is an array, get the order item object instead.
+	if ( is_array( $order_item ) && ! empty( $order_item['id'] ) ) {
+		$order_item = edd_get_order_item( $order_item['id'] );
```

Comment:
This isn't quite right. `$order_item['id']` is actually the product id. https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/payments/class-edd-payment.php#L3270

We might have to think about deliberately adding a new `order_item_id` property so we can reference it correctly.

If we do something like that, we also need to note to update this: https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/payments/functions.php#L883-L901

Reply:
I also think we should get a unit test for this.